configman/value_sources/for_modules.py

```diff
@@ -189,10 +196,40 @@ def local_to_str(a_thing):
 
 
 #==============================================================================
+class OrderableObj(object):
+    """Python3 can't sort non-string types implicitly.
+    """
+    def __init__(self, value):
+        if not isinstance(value, six.string_types):
+            value_str = to_str(value)
+        else:
+            value_str = value
+        self.value_str = value_str
+        self.value = value
+    def __lt__(self, other):
+        return (self.value_str < other.value_str)
+    def __repr__(self):
+        return self.value
+
+
+#==============================================================================
+class OrderableTuple(object):
+    """Python3 can't sort non-string types implicitly.
+    """
+    def __init__(self, value, index=1):
+        self.value = value
+        self.index = index
+    def __lt__(self, other):
+        return (self.value[self.index] < other.value[self.index])
+    def __repr__(self):
+        return self.value
```

Comment:
`total_ordering` lets you define `__lt__` and `__eq__` and then it adds handling for the other comparators based on those. It's a nice shorthand for creating classes that can be compared in various ways.

I spent some time looking into how Python's sort works and whether it's safe to just specify `__lt__`. The source code suggests that it is safe and that it's sufficient to specify only `__lt__`. Then I did some skulking around to see if anyone had asked this question before and whether the Python docs state it's ok and all that and bumped into this:

http://stackoverflow.com/questions/8796886/is-it-safe-to-just-implement-lt-for-a-class-that-will-be-sorted

The summary of that is that it currently works, but it's not something you should rely on working in the future and that the current best practice is to decorate with `functools.total_ordering`.

I'm of the "to hell with Python 2.6" variety and because of that, I think this is a good reason to drop Python 2.6 support. I'm new to configman, so I don't know what the ramifications of that are for the greater configman universe. That's an option.

Another option is to implement it or use a backported library. This is one that my wildly cursory glance suggests is "good enough":

https://pypi.python.org/pypi/total-ordering/0.1.0

An "easy" way to do that is to add something like this to `setup.py`:

def find_install_requires():
    reqs = [x.strip() for x in
            read('requirements.txt').splitlines()
            if x.strip() and not x.startswith('#')]

    try:
        from functools import total_ordering
    except ImportError:
        reqs.append('total-ordering==0.1')
    return reqs
```

I think that'd work, but I'm not 100% sure that works with wheels. We could do a release and then test it out and do a fixit release afterwards if it doesn't work.